### ISSUE TITLE
websocket over http2

### ISSUE THREAD

**Issue by KhafraDev** [nodejs/undici#2458]
Refs: https://github.com/nodejs/undici/pull/2061#issuecomment-1685395999

**Comment by metcoder95** [comment]
Doing a quick check, the feature might not be that hard to add.
We basically have to tweak the first request to conform the WebSocket negotiation spec.

So far, the overall implementation should not need changes; rather just `writeH2`.
Fetch already provides the `mode` through `upgrade` here: https://github.com/nodejs/undici/blob/ac30400a191a78c6148c50e7ff70390a4ea93f90/lib/fetch/index.js#L1963C9-L1963C16

It is a matter of `writeH2` to catch it and attach it with the correct semantics at the headers if the connection is happening through h2, and call `onUpgrade` if everything went well accordingly.
As `fetch` uses the `socket` directly, it should be even less complicated. But we would need to play with it to see if it works.

**Comment by zdm** [comment]
Does this issue planned to be fixed?
It is still actual, I am facing it today, when tried to establish websocket connection with `allowH2 = true`.

**Comment by metcoder95** [comment]
Of course, if there's also interested in contributing, PRs are welcomed
